chat-plugins/mafia.js

```diff
@@ -1,919 +1,931 @@
+			if (!user.can('mute', null, targetRoom) && targetRoom.game.hostid !== user.userid) return user.sendTo(targetRoom, `|error|/mafia close - Access denied.`);
+			if (targetRoom.game.phase !== 'signups') return user.sendTo(targetRoom, `|error|Signups are already closed.`);
+			if (targetRoom.game.playerCount < 2) return user.sendTo(targetRoom, `|error|You need at least 2 players to start.`);
+			targetRoom.game.phase = 'locked';
+			targetRoom.game.sendRoom(targetRoom.game.roomWindow(), {uhtml: true});
+			targetRoom.game.updatePlayers();
+		},
 
-				if (targetObj.settings) {
-					for (let i in defaultSettings) {
-						if (i in targetObj.settings) {
-							settings[i] = !!targetObj.settings[i];
-						} else {
-							settings[i] = defaultSettings[i];
-						}
-					}
-				} else {
-					settings = defaultSettings;
-				}
+		'!closedsetup': true,
+		cs: 'closedsetup',
+		closedsetup: function (target, room, user) {
+			let targetRoom = room;
+			target = target.split(',');
+			if (Rooms(target[0]) && Rooms(target[0]).users[user.userid]) targetRoom = Rooms(target[0]);
+			if (!targetRoom || !targetRoom.game || targetRoom.game.gameid !== 'mafia') return this.errorReply(`There is no game of mafia running in this room.`);
+			if (!user.can('mute', null, targetRoom) && targetRoom.game.hostid !== user.userid) return user.sendTo(targetRoom, `|error|/mafia closedsetup - Access denied.`);
+			let action = targetRoom.id === room.id ? toId(target.join('')) : toId(target[1]);
+			if (!['on', 'off'].includes(action)) return this.parse('/help mafia closedsetup');
+			if (targetRoom.game.started) return user.sendTo(targetRoom, `|error|You can't ${action === 'on' ? 'enable' : 'disable'} closed setup because the game has already started.`);
+			if ((action === 'on' && targetRoom.game.closedSetup) || (action === 'off' && !targetRoom.game.closedSetup)) return user.sendTo(targetRoom, `|error|Closed setup is already ${targetRoom.game.closedSetup ? 'enabled' : 'disabled'}.`);
+			targetRoom.game.closedSetup = action === 'on';
+			targetRoom.game.updateHost();
+		},
+		closesetuphelp: ['/mafia closedsetup [on|off] - '],
+
+		'!reveal': true,
+		reveal: function (target, room, user) {
+			let targetRoom = room;
+			target = target.split(',');
+			if (Rooms(target[0]) && Rooms(target[0]).users[user.userid]) targetRoom = Rooms(target[0]);
+			if (!targetRoom || !targetRoom.game || targetRoom.game.gameid !== 'mafia') return this.errorReply(`There is no game of mafia running in this room.`);
+			if (!user.can('mute', null, targetRoom) && targetRoom.game.hostid !== user.userid) return user.sendTo(targetRoom, `|error|/mafia reveal - Access denied.`);
+			let action = targetRoom.id === room.id ? toId(target.join('')) : toId(target[1]);
+			if (!['on', 'off'].includes(action)) return this.parse('/help mafia reveal');
+			if (targetRoom.game.started) return user.sendTo(targetRoom, `|error|You can't turn revealing of roles ${action} because the game has already started.`);
```

Comment:
I don't really see the point to this, there are some situations when you'd want to imo and allowing it doesnt hurt anything.